Source/d_main.c

```diff
@@ -663,27 +663,62 @@ char *D_DoomPrefDir(void)
 // Calculate the path to the directory for autoloaded WADs/DEHs.
 // Creates the directory as necessary.
 
-static char *autoload_path = NULL;
+static struct {
+    const char *dir;
+    const char *(*func)(void);
+} autoload_basedirs[] = {
+#ifdef WOOFDATADIR
+    {WOOFDATADIR, NULL},
+#endif
+    {NULL, D_DoomPrefDir},
+    {NULL, NULL},
+};
 
-static char *GetAutoloadDir(const char *iwadname, boolean createdir)
+static char **autoload_paths = NULL;
+
+static char *GetAutoloadDir(const char *base, const char *iwadname, boolean createdir)
 {
     char *result;
 
-    if (autoload_path == NULL)
+    result = M_StringJoin(base, DIR_SEPARATOR_S, iwadname, NULL);
+
+    if (createdir)
     {
-        autoload_path = M_StringJoin(D_DoomPrefDir(), DIR_SEPARATOR_S, "autoload", NULL);
+        M_MakeDirectory(result);
     }
 
-    M_MakeDirectory(autoload_path);
+    return result;
+}
 
-    result = M_StringJoin(autoload_path, DIR_SEPARATOR_S, iwadname, NULL);
+static void PrepareAutoloadPaths (void)
+{
+    int i;
 
-    if (createdir)
+    if (autoload_paths)
```

Comment:
I cannot understand this check. Why is it needed and is it always not NULL?

Reply:
It is always NULL before the function was called and always non-NULL after it has been called once. Indeed, it is obsolete now. I had the code structure a bit differently earlier and this is a remnant of this.